### PR TITLE
Update cl_khr_integer_dot_product tests for v2

### PR DESCRIPTION
**Pull request by kpet**
Signed-off-by: Kevin Petit <kevin.petit@arm.com>
Signed-off-by: Marco Cattani <marco.cattani@arm.com>

### CODE REVIEW

**Review by bashbaug — CHANGES_REQUESTED** [review]
Mostly LGTM but I think the checks for acceleration properties need to be guarded by the check for version v2.  Thanks!

File: test_conformance/integer_ops/test_integer_dot_product.cpp [hunk 1]
```diff
@@ -346,19 +361,66 @@ int test_integer_dot_product(cl_device_id deviceID, cl_context context,
...
+    if (extensionVersion >= CL_MAKE_VERSION(2, 0, 0))
+    {
+        test_assert_error(
+            dotCaps & CL_DEVICE_INTEGER_DOT_PRODUCT_INPUT_4x8BIT_KHR,
+            "When cl_khr_integer_dot_product is supported with version >= 2.0.0"
+            "CL_DEVICE_INTEGER_DOT_PRODUCT_INPUT_4x8BIT_KHR must be "
+            "supported");
+    }
+
+    // Check that acceleration properties can be queried
```
[the remark this answers is not in the document]
Comment (kpet) [reply test_conformance/integer_ops/test_integer_dot_product.cpp:381]: Yes, good catch. Now checking the version.

**Review by bashbaug — APPROVED** [review]
Thanks!

**Review by Jeremy-Kemp — APPROVED** [review]
LGTM

**Pull request merged**